js/content/store.js

```diff
@@ -3119,9 +3174,138 @@ let WishlistPageClass = (function(){
+            } else if (e.target.closest(".es_export_modal_close")) {
+                ExtensionLayer.runInPageContext(() => CModal.DismissActiveModal());
+            } else {
+                return;
+            }
+            document.removeEventListener("click", clickListener);
+        }
+
+        function exportWishlist() {
+            ExtensionLayer.runInPageContext(`function(){
+                let data = { rgAllApps: g_Wishlist.rgAllApps, g_rgAppInfo, options: {} };
+                $J("#es_export_form").serializeArray().forEach((item) => data.options[item.name] = item.value);
+                Messenger.postMessage("exportWishlist", JSON.stringify(data));
+            }`);
+            
+            Messenger.addMessageListener("exportWishlist", (data) => {
+                if (!data) { return; }
+                let { rgAllApps, g_rgAppInfo, options } = JSON.parse(data);
+                if (!rgAllApps || !g_rgAppInfo || !options) { return; }
+
+                let toexport;
+                let notes = SyncedStorage.get("user_notes") || {};
+                let json = rgAllApps.map(appid => {
+                    let appinfo = g_rgAppInfo[appid];
+                    appinfo.appid = appid;
+                    if (notes[appid]) {
+                        appinfo.note = notes[appid];
+                    }
+                    return appinfo;
+                });
+
+                if (options.export_type === "JSON") {
+                    toexport = JSON.stringify(json, null, 4);
+                    doExport(options.export_method, toexport);
+                    return;
+                }
+
+                let props = [];
+                let seps = [];
+                for (let key in options) {
+                    if (key.includes("prop")) {
+                        props[+key.replace("prop", "")] = options[key];
+                    }
+                    if (key.includes("sep")) {
+                        seps[+key.replace("sep", "")] = options[key];
+                    }
+                }
+
+                toexport = json.map(app => {
+                    let line = "";
+                    for (let i = 0; i <= props.length; i++) {
+                        if (props[i]) {
+                            line += app[props[i]];
+                        }
+                        if (seps[i]) {
+                            line += seps[i];
+                        }
+                    }
+                    return line;
+                }).join("\n");
+                doExport(options.export_method, toexport);
+            }, true);
+
+            function doExport(method, content) {
+                switch (method) {
+                    case "clipboard":
+                        setClipboard(content)
+                        break;
+                    case "download":
+                        Background.action('download', { content, filename: "wishlist_export.txt" });
```

Comment:
This should have the file extension `.json` if it's a JSON file

Reply:
Good idea